docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+        {
+            //Generate the AutoImplement Attribute
+            const string autoImplementAttributeDeclarationCode = $$"""
+using System;
+namespace {{AutoImplementAttributeNameSpace}};
+
+[AttributeUsage(AttributeTargets.Class, Inherited = false, AllowMultiple = false)]
+sealed class {{AutoImplementAttributeClassName}} : Attribute
+{
+    public string[] InterfacesNames { get; }
+    public {{AutoImplementAttributeClassName}}(params string[] interfacesNames)
+    {
+        InterfacesNames = interfacesNames;
+    }
+}
+""";
+            ctx.AddSource($"{AutoImplementAttributeClassName}.g.cs", autoImplementAttributeDeclarationCode);
+        });
+
+        IncrementalValuesProvider<Model> provider = context.SyntaxProvider.ForAttributeWithMetadataName(
+            fullyQualifiedMetadataName: FullyQualifiedMetadataName,
+            predicate: static (node, cancellationToken_) => node is ClassDeclarationSyntax,
+            transform: static (ctx, cancellationToken) =>
+            {
+                ClassDeclarationSyntax classDeclarationSyntax = (ClassDeclarationSyntax)ctx.TargetNode;
+                
+                string className = classDeclarationSyntax.Identifier.ValueText;
+                string classNameSpace = GetClassNameSpace(classDeclarationSyntax.Parent);
+                InterfaceModel[] interfaces = GetInterfaceModels(ctx.SemanticModel.Compilation, classDeclarationSyntax);
+
+                return new Model(
+                    className,
+                    classNameSpace,
+                    interfaces
+                    );
+            }).Where(m => m is not null);
+
+        context.RegisterSourceOutput(provider.Collect(), Execute);
+    }
+    private static string GetClassNameSpace(SyntaxNode? parent)
+    {
+        return parent is NamespaceDeclarationSyntax namespaceDeclarationSyntax
+            ? namespaceDeclarationSyntax.Name.ToString()
+            : parent is FileScopedNamespaceDeclarationSyntax fileScopedNamespaceDeclarationSyntax
+            ? fileScopedNamespaceDeclarationSyntax.Name.ToString()
+            : AutoImplementAttributeNameSpace;
+    }
+
+    private static InterfaceModel[] GetInterfaceModels(Compilation compilation, ClassDeclarationSyntax classDeclarationSyntax)
+    {
+        //Get class usings in order to build full interface name if needed
+        string[] usings = GetUsings(classDeclarationSyntax.SyntaxTree.GetRoot());
+        string[] interfacesNames = GetInterfacesNames(classDeclarationSyntax.AttributeLists);
+
+        List<InterfaceModel> ret = [];
+        foreach (string interfaceName in interfacesNames)
+        {
+            INamedTypeSymbol? interfaceSymbol = GetInterfaceSymbol(compilation, usings, interfaceName);
+            if (interfaceSymbol is null)
+                continue;
+
+            //Get interface usings, in order to be sure to have the needed usings for the properties that will be generated
```

Comment:
I would not try to copy usings over. Instead, fully qualify everything. There's no telling what will conflict in the rest of the user's file with your own code. It will also not handle interfaces defined in another assembly.